clients/da-vinci-client/src/main/java/com/linkedin/davinci/blob/BlobSnapshotManager.java

```diff
@@ -0,0 +1,116 @@
+package com.linkedin.davinci.blob;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.store.rocksdb.RocksDBUtils;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.rocksdb.Checkpoint;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+
+public class BlobSnapshotManager {
+  private final String basePath;
+
+  @VisibleForTesting
+  protected Checkpoint createCheckpoint(RocksDB rocksDB) {
+    return Checkpoint.create(rocksDB);
+  }
+
+  private static final Logger LOGGER = LogManager.getLogger(BlobSnapshotManager.class);
+  HashMap<String, Map<Integer, AtomicLong>> concurrentSnapshotUsers;
+  HashMap<String, HashMap<Integer, Long>> snapShotTimestamps;
+  Long snapshotRetentionTime;
+
+  // Initialized the snapshot manger for this specific dbDir
+  public BlobSnapshotManager(String basePath, long snapshotRetentionTime) {
```

Comment:
We should instead take the RocksDB object here during initialization, instead of passing it every time when we call the public method below, because I don't expect it will be very handy for the future BlobTransferManger to pass it in every time a transfer request comes in.